### PR TITLE
[Merged by Bors] - chore: golf a messy Matrix proof

### PR DESCRIPTION
**Pull request by eric-wieser**
Using the new `Matrix.ofLinearEquiv` casting function here (modified in this PR to take an explicit argument) makes everything go through cleanly, resolving the porting note.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by j-loreaux** [comment]
bors merge

File: Mathlib/Data/Matrix/Basic.lean
Comment (Vierkantor) [inline Mathlib/Data/Matrix/Basic.lean:244]: I would have gone for defining this in terms of `Matrix.of`, so we don't break through the `Matrix` definition barrier. (And we'd get the `apply` lemmas by `@[simps]`.) On the other hand I don't expect we'll be making `Matrix` an irreducible def / structure soon, so it might be worth having a simpler definition. I'll let you decide.

**Review by Vierkantor — COMMENTED** [review]
bors d+

**Comment by Vierkantor** [comment]
Oh no, race condition!

bors cancel

File: Mathlib/Data/Matrix/Basic.lean
Comment (eric-wieser) [reply Mathlib/Data/Matrix/Basic.lean:244]: I'll wait for https://github.com/leanprover-community/mathlib4/pull/17535 which defines more of these

**Pull request closed without merging**
